proto/osmosis/incentives/tx.proto

```diff
@@ -45,6 +45,14 @@ message MsgCreateGauge {
   // num_epochs_paid_over is the number of epochs distribution will be completed
   // over
   uint64 num_epochs_paid_over = 6;
+
+  // pool_id is the ID of the pool that the gauge is meant to be associated
+  // with. if pool_id is set, then the "QueryCondition.LockQueryType" must be
+  // "NoLock" with all other fields unset, including "QueryCondition.Denom".
+  // However, note that, internally, the empty string ends up being overwritten
+  // with incentivestypes.NoLockExternalGaugeDenom(<pool-id>) so that the
+  // gauges associated with a pool can be queried by this prefix if needed.
+  uint64 pool_id = 7;
```

Comment:
I don't think this is right. Not all fields are unset, cref:

 To create a gauge dedicated to the concentrated liquidity pool, run a `MsgCreateGauge` message in the `x/incentives` module with the following parameters:
- `PoolId`: The ID of the CL pool to create a gauge for.
- `DistrTo.LockQueryType` must be set to `locktypes.LockQueryType.NoLock`
- `DistrTo.Denom` must be an empty string.

The rest of the parameters can be set according to the desired configuration of the gauge. Please read the `x/incentives` module documentation for more information on how to configure gauges.

Reply:
i think "all other fields unset" is talking about the fields specifically in`QueryCondition.LockQueryType`

If LockQueryType = No Lock, you donot have to set the rest of fields including {denom, duration & timestamp}

Reply:
What @stackman27 is saying is correct.

Please see the doc improvements: [52bd732](https://github.com/osmosis-labs/osmosis/pull/5459/commits/52bd732327858632c4bd490a3c155b61c208eae5)